aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+    /// Account executing this operation is not an owner of the multisig account.
+    const ENOT_OWNER: u64 = 2003;
+    /// Transaction payload cannot be empty.
+    const EPAYLOAD_CANNOT_BE_EMPTY: u64 = 4;
+    /// Multisig account must have at least one owner.
+    const ENOT_ENOUGH_OWNERS: u64 = 5;
+    /// Transaction with specified id cannot be found.
+    const ETRANSACTION_NOT_FOUND: u64 = 2006;
+    /// Cannot execute the specified transaction simply via transaction_id as the full payload is not stored on chain.
+    const EPAYLOAD_NOT_STORED: u64 = 7;
+    /// Provided target function does not match the hash stored in the on-chain transaction.
+    const EPAYLOAD_DOES_NOT_MATCH_HASH: u64 = 2008;
+    /// Transaction has not received enough approvals to be executed.
+    const ENOT_ENOUGH_APPROVALS: u64 = 2009;
+    /// Transaction has not received enough rejections to be removed.
+    const ENOT_ENOUGH_REJECTIONS: u64 = 10;
+    /// Number of signatures required must be more than zero and at most the total number of owners.
+    const EINVALID_SIGNATURES_REQUIRED: u64 = 11;
+    /// Payload hash must be exactly 32 bytes (sha3-256).
+    const EINVALID_PAYLOAD_HASH: u64 = 12;
+    /// The multisig account itself cannot be an owner.
+    const EOWNER_CANNOT_BE_MULTISIG_ACCOUNT_ITSELF: u64 = 13;
+    /// Owner has already approved this transaction before.
+    const TRANSACTION_HAS_ALREADY_BEEN_APPROVED: u64 = 14;
+    /// Owner has already rejected this transaction before.
+    const TRANSACTION_HAS_ALREADY_BEEN_REJECTED: u64 = 15;
+
+    /// Indicates the transaction is still pending approvals or rejections.
+    const TRANSACTION_STATUS_PENDING: u64 = 1;
+    /// Indicates the transaction has been executed successfully.
+    const TRANSACTION_STATUS_SUCCEEDED: u64 = 2;
+    /// Indicates the transaction execution has failed. The transaction would still be considered resolved and cannot
+    /// be re-executed.
+    const TRANSACTION_STATUS_FAILED: u64 = 3;
+    /// Transaction has received enough rejections from owners and been removed/skipped.
+    const TRANSACTION_STATUS_REMOVED: u64 = 4;
+
+    /// Represents a multisig account's configurations and transactions.
+    /// This will be stored in the multisig account (created as a resource account separate from any owner accounts).
+    struct MultisigAccount has key {
+        // The list of all owner addresses.
+        owners: vector<address>,
+        // The number of signatures required to pass a transaction (k in k-of-n).
+        signatures_required: u64,
+        // Map from transaction id (incrementing id) to transactions to execute for this multisig account.
+        // Already executed transactions are deleted to save on storage but can always be accessed via events.
+        transactions: Table<u64, MultisigTransaction>,
+        // Last executed or rejected transaction id. Used to enforce in-order executions of proposals.
+        last_transaction_id: u64,
```

Comment:
Naming is a little bit confusing. Sounds like `next_transaction_id == last_transaction_id + 1`, but actually it is not, right?

Reply:
It's not because this allows multiple transactions to be created and remain in pending. Why is this confusing though? Can this be made clearer by making the docs here better?

Reply:
I would personally suggest to rename to 

```
last_executed_transaction_id
next_pending_transaction_id
```

Reply:
Updated. These names are definitely better. Let me know if you have any naming suggestions for other parts.